### DISCUSSION TITLE
Azure SQL Database ?

### DISCUSSION THREAD

**Discussion by kbeaugrand** [microsoft/semantic-kernel#2265] [Ideas]
I saw that SQL database supports cosine distance search in column store. I would like to know if there is some reason to not support Azure SQL database as a vector database in the SK?

If not, I'm ready to implement and test it on the repository...

What do you think?

**Comment by awharrison-28** [comment]
Hi @kbeaugrand looks like you're referring to the feature mentioned in https://learn.microsoft.com/en-us/samples/azure-samples/azure-sql-db-openai/azure-sql-db-openai/? 

We prioritized Azure Cognitive Search (and the PosgreSQL is compatible with Azure CosmosDB for PostgreSQL) to offer connections with Azure services. There is no other reason that it isn't currently supported. We are looking forward to your contribution 😃

**Comment by kbeaugrand** [comment]
> Hi @kbeaugrand looks like you're referring to the feature mentioned in https://learn.microsoft.com/en-us/samples/azure-samples/azure-sql-db-openai/azure-sql-db-openai/?
> 
> We prioritized Azure Cognitive Search (and the PosgreSQL is compatible with Azure CosmosDB for PostgreSQL) to offer connections with Azure services. There is no other reason that it isn't currently supported. We are looking forward to your contribution 😃

Right, I refer to this feature mentioned in the referenced and the sample provided by Davide Mauri.

I've created the feature request in this repo : #2312.

Maybe after my PR a review from @yorek could be interesting to be sure that I used this sample correctly...

**Reply by yorek** [reply]
Hi @kbeaugrand I'd love to review your PR :)

**Reply by kbeaugrand** [reply]
Feel free @yorek : https://github.com/microsoft/semantic-kernel/pull/2339
